server/sandbox_run.go

```diff
@@ -221,10 +221,13 @@ func (s *Server) RunPodSandbox(ctx context.Context, req *pb.RunPodSandboxRequest
 	labels := req.GetConfig().GetLabels()
 
 	// Add special container name label for the infra container
-	labels[types.KubernetesContainerNameLabel] = leaky.PodInfraContainerName
-	labelsJSON, err := json.Marshal(labels)
-	if err != nil {
-		return nil, err
+	labelsJSON := []byte{}
+	if labels != nil {
```

Comment:
Should we create an empty labels and add this conten if labels does not exist

Reply:
If labels is empty/nil, it looks like the empty labelsJSON gets added as an annotation at ~line 330.  I think, but don't know if that's ok.  I was pondering that when I saw your comment.